src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -84,6 +88,22 @@ public IndexStore(DocumentDatabase documentDatabase, ServerStore serverStore)
 
             var stoppedConcurrentIndexBatches = _documentDatabase.Configuration.Indexing.NumberOfConcurrentStoppedBatchesIfRunningLowOnMemory;
             StoppedConcurrentIndexBatches = new SemaphoreSlim(stoppedConcurrentIndexBatches);
+
+            switch (_serverStore.Configuration.Indexing.JournalSynchronizationMode)
+            {
+                case IndexWriteSynchronizationMode.Server:
+                    IndexWritesSynchronization = serverStore.IndexWritesSynchronization;
```

Comment:
Shouldn't we create a new instance here that uses `DatabaseShutdown`, instead? But the same semaphore as the server?